Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -374,11 +374,26 @@ public object Deserialize(IIonReader reader, Type type, IonType ionType)
             }
 
             throw new NotSupportedException($"Data with Ion type {ionType} is not supported for deserialization");
-        }
+        }
+
+        public T Deserialize<T>(IIonReader reader)
+        {
+            try
+            {
+                return (T)Deserialize(reader, typeof(T));
+            }
+            catch (Exception e)
+            {
+                if (options.PermissiveMode && (e is NotSupportedException || e is InvalidCastException))
```

Comment:
Why does this check the exception type, but the other catches above do not?

Reply:
Forgot to catch exception types above. Added.